Framework/DataHandling/src/SaveToSNSHistogramNexus.cpp

```diff
@@ -570,7 +570,7 @@ int SaveToSNSHistogramNexus::WriteGroup(int is_definition) {
               return NX_ERROR;
 
             // Get the X bins
-            const MantidVec &X = inputWorkspace->readX(0);
+            const auto &X = inputWorkspace->y(0);
             // 1 dimension, with that number of bin boundaries
             dataDimensions[0] = static_cast<int>(X.size());
             // The output TOF axis will be whatever size in the workspace.
```

Comment:
- `static_cast<float>(X[i])` on line 581?
- Also consider using `std::copy` for lines 580-581